src/cli/cli.cpp

```diff
@@ -2391,8 +2386,9 @@ otError Interpreter::ProcessPrefixList(void)
 {
     otNetworkDataIterator iterator = OT_NETWORK_DATA_ITERATOR_INIT;
     otBorderRouterConfig  config;
+    otError               error;
 
-    while (otBorderRouterGetNextOnMeshPrefix(mInstance, &iterator, &config) == OT_ERROR_NONE)
+    while ((error = otBorderRouterGetNextOnMeshPrefix(mInstance, &iterator, &config)) == OT_ERROR_NONE)
```

Comment:
Same situation here. I don't think we want to return `error` from `ProcessPrefixList()`. 
The `otBorderRouterGetNextOnMeshPrefix()` returns `OT_ERROR_NOT_FOUND` to indicate it reached end of the list.